src/Executor/Promise/Adapter/SyncPromiseAdapter.php

```diff
@@ -144,13 +186,21 @@ public function wait(Promise $promise)
 
     /**
      * Execute just before starting to run promise completion.
+     *
+     * @template V
+     *
+     * @param Promise<V> $promise
```

Comment:
```suggestion
     * @param Promise<mixed> $promise
```

No need to define a generic type when it is only applied once, that is like defining a variable that is only used once.